src/system.rs

```diff
@@ -754,6 +754,21 @@ macro_rules! system {
                     value: self.value.min(other.value),
                 }
             }
+
+            /// f{32,64}: Restrict a value to a certain interval unless it is NaN.
+            /// Ord:      Restrict a value to a certain interval.
+            #[must_use = "method returns a new number and does not mutate the original value"]
+            #[inline(always)]
+            pub fn clamp(self, min: Self, max: Self) -> Self
+            where
+                V: $crate::num::Float,
+            {
+                Quantity {
+                    dimension: $crate::lib::marker::PhantomData,
+                    units: $crate::lib::marker::PhantomData,
+                    value: self.value.clamp(min.value, max.value),
```

Comment:
This is failing because it doesn't know where the `value.clamp` function is defined for the generic type `V`. The compiler is guessing the `Ord` trait because it is the only one with a `clamp` method. I see a couple solutions:

 * Use https://docs.rs/num/latest/num/fn.clamp.html. This may not be as efficient for `f32`/`f64`.
 * Create another `storage_types!` block just for float and then the code will compile

Reply:
If you're going to raise the MSRV to 1.50.0, then I'd rather couple to the standard behaviour.

> Create another `storage_types!` block just for float 

You mean, alongside the one which already exists for `Float` and `Complex`?

If so, I'm getting something that compiles, but fails tests because of panics in the case that `max` < `min`. How do you put constraints on what quickcheck generates?

Reply:
Both my suggestions were in the context of raising the MSRV to 1.50. The first, using `num::clamp`, could be implemented in the `impl<D, U, V> Quantity<D, U, V> ...` block and would apply to all underlying storage types, however because it's using `num::clamp` it may not be as performant as a direct `f32::clamp` call. The second suggestion would be another `storage_types!` block that is just for `float` (the existing can't be used because `Complex` doesn't have a `clamp` method) and would essentially proxy `f32`/`f64` `clamp`. I lean towards the second option.

Reject invalid test input using `TestResult::discard();`:
https://github.com/iliekturtles/uom/blob/7c4b27b3cdc9530acd0c7116d5e073c1058207ce/src/tests/system.rs#L112-L114

Reply:
Should I add some `#[should_panic]` tests? If so, I'm not sure how these fit in to the test strucre.

Reply:
Sure, something like `plant.tblResourcePackagingLinePackages` with `#[should_panic]` can be added as a non-quickcheck test up around line 263.